compiler/transforms/insert_sync_barrier.py

```diff
@@ -34,6 +34,10 @@ def match_and_rewrite(self, module: builtin.ModuleOp, rewriter: PatternRewriter)
                 # clear the list
                 ops_to_sync = []
 
+            if isinstance(op, snax.ClusterSyncOp):
+                # synchronisation ok, clear list
+                ops_to_sync = []
+
```

Comment:
just to make my test work 😬 . The test is already mid-way in the pipeline, and thus has some tests applied 2x. Without this, it tries to add multiple synchronizations. Because some regions are already dispatched, this leads to some synchronization bugs.

I guess we should make sure applying a pass at the wrong time / applying it twice should never result in an error. This makes sure this is the case for this one.